js/bittrex.js

```diff
@@ -659,13 +661,26 @@ module.exports = class bittrex extends Exchange {
         if (since !== undefined) {
             const now = this.milliseconds ();
             const difference = Math.abs (now - since);
-            if (difference > 86400000) {
+            const date = this.ymd (since);
+            const parts = date.split ('-');
+            const year = this.safeInteger (parts, 0);
+            const month = this.safeInteger (parts, 1);
+            const day = this.safeInteger (parts, 2);
+            const curr_date = this.ymd (now);
+            const curr_parts = curr_date.split ('-');
+            const curr_year = this.safeInteger (curr_parts, 0);
+            const curr_month = this.safeInteger (curr_parts, 1);
+            if (timeframe === '1d' && year !== curr_year) {
+                // Current year is provided by the "current" endpoint
```

Comment:
@xmatthias can you plz comment a bit on this line, what is meant under the "current" endpoint here?

Reply:
As far as i see from their docs, they currently have two endpoints, namely:

- A _"recent"_ endpoint https://bittrex.github.io/api/v3#operation--markets--marketSymbol--candles--candleInterval--recent-get
- And a _"historical"_ endpoint https://bittrex.github.io/api/v3#operation--markets--marketSymbol--candles--candleInterval--historical--year---month---day--get

However, it's not entirely clear what "recent" and "historical" refer to.

Reply:
I think the key for this is in the changelog: https://bittrex.github.io/api/v3#topic-Change-Log (From 05/27/2020).

Basically, the historical endpoint provides history (excluding current month (for 1h) or current year (for 1d)) candles.
For current year / current month (1h) - the "recent" endpoint should be used.

clearly, my "spelling" is wrong - where it says "current" endpoint, it's really "recent" endpoint (must've mixed that up in my mind) (i've fixed this now).

Reply:
@xmatthias alright, thanks for clarifying it for me ) I'll add the _"recent"_ family of endpoints and will make a few minor edits for it, will merge today! Thx again!

Reply:
i thought `v3publicGetMarketsMarketSymbolCandles` would cover this? (this does work fine to get the latest candles - and is what the above logic falls back to if the timerange is determined to be "recent").


also major edits are good - my javascript is improving - so i'll for sure look at your edits to learn 👍

Reply:
> i thought v3publicGetMarketsMarketSymbolCandles would cover this? (this does work fine to get the latest candles - and is what the above logic falls back to if the timerange is determined to be "recent").

Yep, it works, but according to their doc it may go down anytime now, so we'd better move to their new schema entirely.